src/main/kotlin/com/openlattice/assembler/AssemblerConnectionManager.kt

```diff
@@ -235,53 +222,67 @@ class AssemblerConnectionManager(
 
         val securablePrincipalsToAdd = securePrincipalsManager.getSecurablePrincipals(validUserPrincipals)
         if (securablePrincipalsToAdd.isNotEmpty()) {
-            val userNames = securablePrincipalsToAdd.map { dbCredentialService.getDbUsername(buildPostgresUsername(it)) }
-            configureUsersInDatabase(dataSource, dbName, userNames)
+            val userNames = securablePrincipalsToAdd.map { dbCredentialService.getDbUsername(it) }
+            configureUsersInDatabase(dataSource, organizationId, userNames)
         }
     }
 
     fun addMembersToOrganization(
-            dbName: String,
+            organizationId: UUID,
+            authorizedPropertyTypesOfEntitySetsByPrincipal: Map<SecurablePrincipal, Map<EntitySet, Collection<PropertyType>>>
+    ) {
+        extDbManager.connectToOrg(organizationId).let { dataSource ->
+            addMembersToOrganization(organizationId, dataSource, authorizedPropertyTypesOfEntitySetsByPrincipal)
+        }
+    }
+
+    fun addMembersToOrganization(
+            organizationId: UUID,
             dataSource: HikariDataSource,
             authorizedPropertyTypesOfEntitySetsByPrincipal: Map<SecurablePrincipal, Map<EntitySet, Collection<PropertyType>>>
     ) {
         if (authorizedPropertyTypesOfEntitySetsByPrincipal.isNotEmpty()) {
             val authorizedPropertyTypesOfEntitySetsByPostgresUser = authorizedPropertyTypesOfEntitySetsByPrincipal
-                    .mapKeys { dbCredentialService.getDbUsername(buildPostgresUsername(it.key)) }
+                    .mapKeys { dbCredentialService.getDbUsername(it.key) }
             val userNames = authorizedPropertyTypesOfEntitySetsByPostgresUser.keys
-            configureUsersInDatabase(dataSource, dbName, userNames)
+            configureUsersInDatabase(dataSource, organizationId, userNames)
             dataSource.connection.use { connection ->
                 grantSelectForNewMembers(connection, authorizedPropertyTypesOfEntitySetsByPostgresUser)
             }
         }
     }
 
     fun removeMembersFromOrganization(
-            dbName: String,
+            organizationId: UUID,
+            principals: Collection<SecurablePrincipal>
+    ) {
+        extDbManager.connectToOrg(organizationId).let { dataSource ->
+            removeMembersFromOrganization( organizationId, dataSource, principals )
+        }
+    }
+
+    fun removeMembersFromOrganization(
+            organizationId: UUID,
             dataSource: HikariDataSource,
             principals: Collection<SecurablePrincipal>
     ) {
         if (principals.isNotEmpty()) {
-            val userNames = principals.map { dbCredentialService.getDbUsername(buildPostgresUsername(it)) }
-            revokeConnectAndSchemaUsage(dataSource, dbName, userNames)
+            val userNames = principals.map { dbCredentialService.getDbUsername(it) }
```

Comment:
seems ideal to do this as a bulk `getDbUsernames(principals)` call but it probably isn't that important since this will probably almost always be called on one user at a time (at at the very least, it won't be like tens of thousands or anything like that)

Reply:
👍  definitely should bulk

Reply:
although this is also more code I haven't functionally changed

Reply:
https://jira.openlattice.com/browse/LATTICE-2576